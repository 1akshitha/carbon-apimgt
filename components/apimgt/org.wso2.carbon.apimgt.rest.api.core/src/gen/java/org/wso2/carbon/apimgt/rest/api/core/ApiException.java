package org.wso2.carbon.apimgt.rest.api.core;

@javax.annotation.Generated(value = "class org.wso2.maven.plugins.JavaMSF4JServerCodegen", date = "2016-12-08T17:11:20.964+05:30")
public class ApiException extends Exception{
    private int code;
    public ApiException (int code, String msg) {
        super(msg);
        this.code = code;
    }
}
