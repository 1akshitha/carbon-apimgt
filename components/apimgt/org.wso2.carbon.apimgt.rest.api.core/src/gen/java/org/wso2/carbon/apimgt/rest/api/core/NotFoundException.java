package org.wso2.carbon.apimgt.rest.api.core;

@javax.annotation.Generated(value = "class org.wso2.maven.plugins.JavaMSF4JServerCodegen", date = "2016-12-08T17:11:20.964+05:30")
public class NotFoundException extends ApiException {
    private int code;
    public NotFoundException (int code, String msg) {
        super(code, msg);
        this.code = code;
    }
}
