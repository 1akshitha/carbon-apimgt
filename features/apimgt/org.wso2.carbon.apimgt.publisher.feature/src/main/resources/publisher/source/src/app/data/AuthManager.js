/*
 * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 * WSO2 Inc. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
"use strict";

import axios from 'axios'
import qs from 'qs'
import Utils from './Utils'
import User from './User'
import APIClient from './APIClient'
import APIClientFactory from "./APIClientFactory";

class AuthManager {
    constructor() {
        this.isLogged = false;
        this.username = null;
    }

    /**
     * Refresh the access token and set new access token to the intercepted request
     * @param {Request} request
     */
    static refreshTokenOnExpire(request) {
        const refreshPeriod = 60;
        const user = AuthManager.getUser();
        let timeToExpire = Utils.timeDifference(user.getExpiryTime());
        if (timeToExpire >= refreshPeriod) {
            return request;
        }
        let loginPromise = AuthManager.refresh();
        loginPromise.then(response => {
            const user = AuthManager.loginUserMapper(response);
            AuthManager.setUser(user);
        });
        loginPromise.catch(
            function(error) {
                let error_data = JSON.parse(error.responseText);
                let message = "Error while refreshing token You will be redirect to the login page ...";
                console.error(message);
            }
        );
    }

    /**
     * Static method to handle unauthorized user action error catch, It will look for response status code and skip !401 errors
     * @param {object} error_response
     */
    static unauthorizedErrorHandler(error_response) {
        if (error_response.status !== 401) { /* Skip unrelated response code to handle in unauthorizedErrorHandler*/
            throw error_response;
            /* re throwing the error since we don't handle it here and propagate to downstream error handlers in catch chain*/
        }
        let message = "The session has expired" + ".<br/> You will be redirect to the login page ...";
        if (typeof noty !== 'undefined') {
            noty({
                text: message,
                type: 'error',
                dismissQueue: true,
                modal: true,
                progressBar: true,
                timeout: 5000,
                layout: 'top',
                theme: 'relax',
                maxVisible: 10,
                callback: {
                    afterClose: function() {
                        window.location = loginPageUri;
                    },
                }
            });
        } else {
            throw error_response;
        }
    }

    /**
     * An user object is return in present of user logged in user info in browser local storage, at the same time checks for partialToken in the cookie as well.
     * This may give a partial indication(passive check not actually check the token validity via an API) of whether the user has logged in or not, The actual API call may get denied
     * if the cookie stored access token is invalid/expired
     * @param {string} environmentName: label of the environment, the user to be retrieved from
     * @returns {User | null} Is any user has logged in or not
     */
    static getUser() {
        const userData = localStorage.getItem(`${User.CONST.LOCALSTORAGE_USER}_${Utils.getEnvironment().label}`);
        const environmentName = "_" + Utils.getEnvironment().label;
        const partialToken = Utils.getCookie(User.CONST.WSO2_AM_TOKEN_1 + environmentName);
        if (!(userData && partialToken)) {
            return null;
        }
        return User.fromJson(JSON.parse(userData));
    }

    /**
     * Persist an user in browser local storage and in-memory, Since only one use can be logged into the application at a time,
     * This method will override any previously persist user data.
     * @param {User} user : An instance of the {User} class
     * @param {string} environmentName: label of the environment to be set the user
     */
    static setUser(user, environmentName) {
        environmentName = environmentName || Utils.getEnvironment().label;
        if (!user instanceof User) {
            throw new Error("Invalid user object");
        }
        if (user) {
            localStorage.setItem(`${User.CONST.LOCALSTORAGE_USER}_${Utils.getEnvironment().label}`,
                JSON.stringify(user.toJson()));
        }
    }

    static hasScopes(resourcePath, resourceMethod) {
        let userscopes = this.getUser().scopes;
        let validScope = APIClient.getScopeForResource(resourcePath, resourceMethod);
        return validScope.then(scope => {
            return userscopes.includes(scope)
        });
    }

    /**
     * By given username and password Authenticate the user, Since this REST API has no swagger definition,
     * Can't use swaggerjs to generate client.Hence using Axios to make AJAX calls
     * @param {String} username : Username of the user
     * @param {String} password : Plain text password
     * @param {Object} environment : environment object
     * @returns {AxiosPromise} : Promise object with the login request made
     */
    authenticateUser(username, password, environment) {
        const headers = {
            'Accept': 'application/json',
            'Content-Type': 'application/x-www-form-urlencoded'
        };
        const data = {
            username: username,
            password: password,
            grant_type: 'password',
            validity_period: -1,
            scopes: AuthManager.CONST.USER_SCOPES,
            remember_me: true // By default always remember user session
        };
        let promised_response = axios(Utils.getLoginTokenPath(environment), {
            method: "POST",
            data: qs.stringify(data),
            headers: headers,
            withCredentials: true
        });
        //Set the environment that user tried to authenticate
        let previous_environment = Utils.getEnvironment();
        Utils.setEnvironment(environment);

        promised_response.then(response => {
            const user = AuthManager.loginUserMapper(response);
            AuthManager.setUser(user);
            this.setupAutoRefresh();
        }).catch(error => {
            console.error("Authentication Error:\n", error);
            Utils.setEnvironment(previous_environment);
        });
        return promised_response;
    }

    /**
     * Return an user object given the login request response object
     * @param {Object} response Response object received from either Axios or Fetch libraries
     * @returns {User} Instance of an user who is currently logged in (for the selected environment)
     */
    static loginUserMapper(response) {
        let data = response.data;
        const validityPeriod = data.validityPeriod; // In seconds
        const WSO2_AM_TOKEN_1 = data.partialToken;
        const user = new User(Utils.getEnvironment().label, data.authUser, data.idToken);
        user.setPartialToken(WSO2_AM_TOKEN_1, validityPeriod, Utils.CONST.CONTEXT_PATH);
        user.setExpiryTime(validityPeriod);
        user.scopes = data.scopes.split(" ");
        return user;
    }

    /**
     * Revoke the issued OAuth access token for currently logged in user and clear both cookie and localstorage data.
     */
    logout() {
        let authHeader = "Bearer " + AuthManager.getUser().getPartialToken();
        //TODO Will have to change the logout end point url to contain the app context(i.e. publisher/store, etc.)
        let url = Utils.getAppLogoutURL();
        let headers = {
            'Accept': 'application/json',
            'Content-Type': 'application/json',
            'Authorization': authHeader
        };
        const promisedLogout = axios.post(url, null, { headers: headers });
        return promisedLogout.then(response => {
            Utils.delete_cookie(User.CONST.WSO2_AM_TOKEN_1, Utils.CONST.CONTEXT_PATH);
            localStorage.removeItem(User.CONST.LOCALSTORAGE_USER);
            new APIClientFactory().destroyAPIClient(Utils.getEnvironment().label); // Single client should be re initialize after log out
        });
    }

    setupAutoRefresh() {
        const user = AuthManager.getUser();
        const bufferTime = 1000 * 10; // Give 10 sec buffer time before token expire, considering the network delays and ect.
        let triggerIn = Utils.timeDifference(user.getExpiryTime() - bufferTime);
        if (user) {
            setTimeout(AuthManager.refreshTokenOnExpire, triggerIn * 1000);
        } else {
            throw new Error("No user exist for current session! Needs to login before setting up refresh");
        }
    }

    static refresh() {
        const authHeader = "Bearer " + AuthManager.getUser().getRefreshPartialToken();
        let params = {
            grant_type: 'refresh_token',
            validity_period: -1,
            scopes: AuthManager.CONST.USER_SCOPES
        };
        let referrer = (document.referrer.indexOf("https") !== -1) ? document.referrer : null;
        let url = '/login/token' + Utils.CONST.CONTEXT_PATH;
        /* TODO: Fetch this from configs ~tmkb*/
        let headers = {
            'Authorization': authHeader,
            'Accept': 'application/json',
            'Content-Type': 'application/x-www-form-urlencoded',
            'X-Alt-Referer': referrer
        };
        return axios.post(url, qs.stringify(params), { headers: headers });
    }

}

// TODO: derive this from swagger definitions ~tmkb
AuthManager.CONST = {
    USER_SCOPES: "apim:api_view apim:api_create apim:api_publish apim:tier_view apim:tier_manage " +
        "apim:subscription_view apim:subscription_block apim:subscribe apim:external_services_discover"
};
export default AuthManager;