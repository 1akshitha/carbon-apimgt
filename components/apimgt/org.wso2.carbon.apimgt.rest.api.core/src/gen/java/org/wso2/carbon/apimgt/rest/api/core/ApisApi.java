package org.wso2.carbon.apimgt.rest.api.core;

import org.wso2.carbon.apimgt.rest.api.core.factories.ApisApiServiceFactory;

import io.swagger.annotations.ApiParam;

import org.wso2.carbon.apimgt.rest.api.core.dto.ErrorDTO;

import org.wso2.msf4j.Microservice;
import org.wso2.msf4j.Request;
import org.osgi.service.component.annotations.Component;

import java.io.InputStream;
import javax.ws.rs.core.Context;
import javax.ws.rs.core.Response;
import javax.ws.rs.*;

@Component(
    name = "org.wso2.carbon.apimgt.rest.api.core.ApisApi",
    service = Microservice.class,
    immediate = true
)
@Path("/api/am/core/v1.[\\d]+/apis")
@Consumes({ "application/json" })
@Produces({ "application/json" })
@io.swagger.annotations.Api(description = "the apis API")
@javax.annotation.Generated(value = "org.wso2.maven.plugins.JavaMSF4JServerCodegen", date = "2017-04-26T22:35:57.497+05:30")
public class ApisApi implements Microservice  {
   private final ApisApiService delegate = ApisApiServiceFactory.getApisApi();

    @GET
    @Path("/{apiId}/gateway-config")
    @Consumes({ "application/json" })
    @Produces({ "text/plain" })
    @io.swagger.annotations.ApiOperation(value = "Get gateway definition", notes = "This operation can be used to retrieve the gateway configuration of an API. ", response = void.class, tags={ "API (Individual)", })
    @io.swagger.annotations.ApiResponses(value = { 
        @io.swagger.annotations.ApiResponse(code = 200, message = "OK. Requested gateway configuration of the API is returned ", response = void.class),
        
        @io.swagger.annotations.ApiResponse(code = 304, message = "Not Modified. Empty body because the client has already the latest version of the requested resource (Will be supported in future). ", response = void.class),
        
        @io.swagger.annotations.ApiResponse(code = 404, message = "Not Found. Requested API does not exist. ", response = void.class),
        
        @io.swagger.annotations.ApiResponse(code = 406, message = "Not Acceptable. The requested media type is not supported ", response = void.class) })
    public Response apisApiIdGatewayConfigGet(@ApiParam(value = "The UUID of an API ",required=true) @PathParam("apiId") String apiId
,@ApiParam(value = "Media types acceptable for the response. Default is application/json. " , defaultValue="application/json")@HeaderParam("Accept") String accept
,@ApiParam(value = "Validator for conditional requests; based on the ETag of the formerly retrieved variant of the resourec. " )@HeaderParam("If-None-Match") String ifNoneMatch
,@ApiParam(value = "Validator for conditional requests; based on Last Modified header of the formerly retrieved variant of the resource. " )@HeaderParam("If-Modified-Since") String ifModifiedSince
, @Context Request request)
    throws NotFoundException {
        return delegate.apisApiIdGatewayConfigGet(apiId,accept,ifNoneMatch,ifModifiedSince, request);
    }
}
