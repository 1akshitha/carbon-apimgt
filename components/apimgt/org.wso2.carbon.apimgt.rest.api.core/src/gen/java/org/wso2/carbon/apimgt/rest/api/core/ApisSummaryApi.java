package org.wso2.carbon.apimgt.rest.api.core;

import org.wso2.carbon.apimgt.rest.api.core.factories.ApisSummaryApiServiceFactory;

import io.swagger.annotations.ApiParam;

import org.wso2.carbon.apimgt.rest.api.core.dto.APISummaryListDTO;
import org.wso2.carbon.apimgt.rest.api.core.dto.ErrorDTO;

import org.wso2.msf4j.Microservice;
import org.wso2.msf4j.Request;
import org.osgi.service.component.annotations.Component;

import java.io.InputStream;
import javax.ws.rs.core.Context;
import javax.ws.rs.core.Response;
import javax.ws.rs.*;

@Component(
    name = "org.wso2.carbon.apimgt.rest.api.core.ApisSummaryApi",
    service = Microservice.class,
    immediate = true
)
@Path("/api/am/core/v1.[\\d]+/apis-summary")
@Consumes({ "application/json" })
@Produces({ "application/json" })
@io.swagger.annotations.Api(description = "the apis-summary API")
@javax.annotation.Generated(value = "org.wso2.maven.plugins.JavaMSF4JServerCodegen", date = "2017-04-26T22:35:57.497+05:30")
public class ApisSummaryApi implements Microservice  {
   private final ApisSummaryApiService delegate = ApisSummaryApiServiceFactory.getApisSummaryApi();

    @GET
    
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    @io.swagger.annotations.ApiOperation(value = "Summary of all available apis", notes = "Request summary of all available apis.", response = APISummaryListDTO.class, tags={  })
    @io.swagger.annotations.ApiResponses(value = { 
        @io.swagger.annotations.ApiResponse(code = 200, message = "OK. Subscription Validation Information. ", response = APISummaryListDTO.class),
        
        @io.swagger.annotations.ApiResponse(code = 400, message = "Bad Request. Invalid request or validation error ", response = APISummaryListDTO.class),
        
        @io.swagger.annotations.ApiResponse(code = 404, message = "Not Found. The resource to be updated does not exist. ", response = APISummaryListDTO.class),
        
        @io.swagger.annotations.ApiResponse(code = 406, message = "Not Acceptable. The requested media type is not supported ", response = APISummaryListDTO.class) })
    public Response apisSummaryGet(@ApiParam(value = "Media types acceptable for the response. Default is application/json. " , defaultValue="application/json")@HeaderParam("Accept") String accept
,@ApiParam(value = "Validator for conditional requests; based on the ETag of the formerly retrieved variant of the resourec. " )@HeaderParam("If-None-Match") String ifNoneMatch
,@ApiParam(value = "Validator for conditional requests; based on Last Modified header of the formerly retrieved variant of the resource. " )@HeaderParam("If-Modified-Since") String ifModifiedSince
, @Context Request request)
    throws NotFoundException {
        return delegate.apisSummaryGet(accept,ifNoneMatch,ifModifiedSince, request);
    }
}
