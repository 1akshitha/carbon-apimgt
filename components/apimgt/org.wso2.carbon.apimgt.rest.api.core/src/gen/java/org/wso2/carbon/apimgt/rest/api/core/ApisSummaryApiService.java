package org.wso2.carbon.apimgt.rest.api.core;

import javax.ws.rs.core.Response;

@javax.annotation.Generated(value = "class org.wso2.maven.plugins.JavaMSF4JServerCodegen", date = "2017-02-09T14:55:54.962+05:30")
public abstract class ApisSummaryApiService {
    public abstract Response apisSummaryGet() throws NotFoundException;
}
