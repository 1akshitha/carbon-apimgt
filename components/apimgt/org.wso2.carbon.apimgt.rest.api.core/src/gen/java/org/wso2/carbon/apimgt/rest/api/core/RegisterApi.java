package org.wso2.carbon.apimgt.rest.api.core;

import org.wso2.carbon.apimgt.rest.api.core.factories.RegisterApiServiceFactory;

import io.swagger.annotations.ApiParam;

import org.wso2.carbon.apimgt.rest.api.core.dto.ErrorDTO;
import org.wso2.carbon.apimgt.rest.api.core.dto.RegistrationDTO;
import org.wso2.carbon.apimgt.rest.api.core.dto.RegistrationSummaryDTO;

import org.wso2.msf4j.Microservice;
import org.osgi.service.component.annotations.Component;

import java.io.InputStream;

import org.wso2.msf4j.formparam.FormDataParam;
import org.wso2.msf4j.formparam.FileInfo;

import javax.ws.rs.core.Response;
import javax.ws.rs.*;

@Component(
    name = "org.wso2.carbon.apimgt.rest.api.core.RegisterApi",
    service = Microservice.class,
    immediate = true
)
@Path("/api/am/core/v1/register")
@Consumes({ "application/json" })
@Produces({ "application/json" })
@io.swagger.annotations.Api(description = "the register API")
@javax.annotation.Generated(value = "org.wso2.maven.plugins.JavaMSF4JServerCodegen", date = "2017-03-31T16:40:30.481+05:30")
public class RegisterApi implements Microservice  {
   private final RegisterApiService delegate = RegisterApiServiceFactory.getRegisterApi();

    @POST
    
    @Consumes({ "application/json" })
    @Produces({ "application/json" })
    @io.swagger.annotations.ApiOperation(value = "Register a Gateway", notes = "This operation can be used to register a gateway. ", response = RegistrationSummaryDTO.class, tags={ "Register Gateway", })
    @io.swagger.annotations.ApiResponses(value = { 
        @io.swagger.annotations.ApiResponse(code = 201, message = "Created. Successful response with Registration Summary details in the body. ", response = RegistrationSummaryDTO.class),
        
        @io.swagger.annotations.ApiResponse(code = 400, message = "Bad Request. Invalid request or validation error. ", response = RegistrationSummaryDTO.class),
        
        @io.swagger.annotations.ApiResponse(code = 415, message = "Unsupported Media Type. The entity of the request was in a not supported format. ", response = RegistrationSummaryDTO.class) })
    public Response registerPost(@ApiParam(value = "Register object that needs to be added " ,required=true) RegistrationDTO body
,@ApiParam(value = "Media type of the entity in the body. Default is JSON. " ,required=true, defaultValue="JSON")@HeaderParam("Content-Type") String contentType
)
    throws NotFoundException {
        return delegate.registerPost(body,contentType);
    }
}
