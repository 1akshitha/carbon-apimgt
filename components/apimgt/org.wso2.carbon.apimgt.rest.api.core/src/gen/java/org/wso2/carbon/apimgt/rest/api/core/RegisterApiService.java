package org.wso2.carbon.apimgt.rest.api.core;

import org.wso2.carbon.apimgt.rest.api.core.*;
import org.wso2.carbon.apimgt.rest.api.core.dto.*;

import org.wso2.msf4j.formparam.FormDataParam;
import org.wso2.msf4j.formparam.FileInfo;

import org.wso2.carbon.apimgt.rest.api.core.dto.ErrorDTO;
import org.wso2.carbon.apimgt.rest.api.core.dto.RegistrationDTO;
import org.wso2.carbon.apimgt.rest.api.core.dto.RegistrationSummaryDTO;

import java.util.List;
import org.wso2.carbon.apimgt.rest.api.core.NotFoundException;

import java.io.InputStream;

import javax.ws.rs.core.Response;
import javax.ws.rs.core.SecurityContext;

@javax.annotation.Generated(value = "org.wso2.maven.plugins.JavaMSF4JServerCodegen", date = "2017-03-31T16:40:30.481+05:30")
public abstract class RegisterApiService {
    public abstract Response registerPost(RegistrationDTO body
 ,String contentType
 ) throws NotFoundException;
}
