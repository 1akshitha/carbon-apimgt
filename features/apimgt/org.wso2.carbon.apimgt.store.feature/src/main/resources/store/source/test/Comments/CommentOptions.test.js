/*
 * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 * WSO2 Inc. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import { unwrap } from '@material-ui/core/test-utils';
import { Typography } from '@material-ui/core';
import CommentOptions from '../../src/app/components/Apis/Details/Comments/CommentOptions';
import AuthManager from '../../src/app/data/AuthManager';

const CommentOptionsUnwrapped = unwrap(CommentOptions);
let reply; let
    comment;
const user = {
    expiryTime: '2018-12-16T17:38:24.193Z',
    name: 'admin',
    remember: false,
    scopes: ['openid', 'apim:api_delete', 'apim:api_publish', 'apim:external_services_discover'],
};

/**
 * Initialize common properties to be passed
 * @param {*} props properies to be override
 */
function createTestProps(props) {
    reply = {
        commentId: 'adf03093-74a3-4cd3-b5d0-a30d32a90f4b',
        category: 'General',
        parentCommentId: 'ebf03093-74a3-4cd3-b5d0-a30d32a90f4b',
        username: 'admin',
        commentText: 'My new comment reply',
        createdTime: '2018-09-27T10:17:44.444Z',
        createdBy: 'admin',
        lastUpdatedTime: '2018-09-27T11:37:03.570Z',
        lastUpdatedBy: 'admin',
        replies: [],
    };

    comment = {
        commentId: 'ebf03093-74a3-4cd3-b5d0-a30d32a90f4b',
        category: 'General',
        parentCommentId: null,
        username: 'admin',
        commentText: 'My new comment',
        createdTime: '2018-09-27T10:16:44.444Z',
        createdBy: 'admin',
        lastUpdatedTime: '2018-09-27T10:37:03.570Z',
        lastUpdatedBy: 'admin',
        replies: [reply],
    };

    return {
        classes: {},
        index: 0,
        comment,
        editIndex: -1,
        handleClickOpen: jest.fn(),
        showEditComment: jest.fn(),
        theme: { custom: { maxCommentLength: 1300 } },
        ...props,
    };
}

let wrapper;
const props = createTestProps();

beforeEach(() => {
    wrapper = shallow(<CommentOptionsUnwrapped {...props} />);
});

describe('<CommentOptions /> rendering', () => {
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a <Typography /> to type the comment', () => {
        AuthManager.setUser(user, 'environment');
        console.log(AuthManager.getUser());
        expect(wrapper.find(Typography)).toHaveLength(1);
    });
});
