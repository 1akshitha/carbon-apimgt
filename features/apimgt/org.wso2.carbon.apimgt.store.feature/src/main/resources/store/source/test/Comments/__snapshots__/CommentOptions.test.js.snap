// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentOptions /> rendering renders correctly 1`] = `undefined`;
